netty/src/main/java/io/grpc/transport/netty/ProtocolNegotiators.java

```diff
@@ -224,10 +245,13 @@ public void flush(ChannelHandlerContext ctx) {
 
     @Override
     public void close(ChannelHandlerContext ctx, ChannelPromise future) throws Exception {
-      failBufferedAndClose(ctx);
+      fail(ctx, new Exception("Channel closed while performing protocol negotiation"));
```

Comment:
The old code would call the approximate equivalent of super.close here. Closing within this class is important, as it is very hard to track the closure back to the negotiationFailed implementation.